src/main/java/org/jenkinsci/plugins/github/pullrequest/builders/GitHubPRStatusBuilder.java

```diff
@@ -52,41 +57,52 @@ public GitHubPRMessage getStatusMessage() {
     }
 
     @Override
-    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener)
-            throws InterruptedException, IOException {
-        GitHubPRTrigger trigger = build.getProject().getTrigger(GitHubPRTrigger.class);
+    public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull Launcher launcher,
+                        @Nonnull TaskListener listener) throws InterruptedException, IOException {
+        // No triggers in Run class, but we need it
+        final GitHubPRTrigger trigger = JobInfoHelpers.triggerFrom(run.getParent(), GitHubPRTrigger.class);
         if (trigger == null) {
-            return true;
+            // silently skip. TODO implement error handler, like in publishers
+            return;
         }
 
-        GitHubPRCause cause = build.getCause(GitHubPRCause.class);
+        GitHubPRCause cause = run.getCause(GitHubPRCause.class);
         if (cause == null) {
-            return true;
+            return;
         }
 
         // GitHub status for commit
         try {
             if (statusMessage != null) {
-                String url = trigger.getDescriptor().getJenkinsURL() + build.getUrl();
-
-                trigger.getRemoteRepo().createCommitStatus(cause.getHeadSha(),
+                String url = trigger.getDescriptor().getJenkinsURL() + run.getUrl();
+                String expandedMessage;
+                if (run instanceof AbstractBuild<?, ?>) {
+                    final AbstractBuild<?, ?> build = (AbstractBuild<?, ?>) run;
+                    expandedMessage = statusMessage.expandAll(build, listener);
+                } else {
+                    // Workflow doesn't expand variables with token macro!
+                    expandedMessage = statusMessage.getContent();
```

Comment:
Sounds like i should wrote Util that will try expand all possible variables from JobMixIns. That will at least expand ParameterActions.